driver/native/native.go

```diff
@@ -249,20 +281,19 @@ func (d *Driver) Parse(rctx context.Context, src string) (nodes.Node, error) {
 		Content: str, Encoding: d.ec,
 	})
 	if err != nil {
-		// Cannot write data - this means the stream is broken or driver crashed.
-		// We will try to recover by reading the response, but since it might be
-		// a stack trace or an error message, we will read it as a "raw" value.
-		// This preserves an original text instead of failing with decoding error.
-		var raw json.RawMessage
-		// TODO: this reads a single line only; we can be smarter and read the whole log if driver cannot recover
-		if err := d.dec.Decode(&raw); err != nil {
-			// stream is broken on both sides, cannot get additional info
-			return nil, driver.ErrDriverFailure.Wrap(err)
-		}
-		return nil, driver.ErrDriverFailure.Wrap(fmt.Errorf("error: %v; %s", err, string(raw)))
+		return nil, err
 	}
 
 	r, err := d.readResponse(ctx)
+	if err == io.EOF {
+		// driver just died; try to restart it once
```

Comment:
Could be interesting, maybe for future versions, to wait a little before restarting it and increase that wait on N restart tries until failing completely.

Reply:
Good idea! Added a note. `bblfshd` already has this logic, but it may be useful here as well.